### ISSUE TITLE
"License: Unspecified" due to unindexable file?

### ISSUE THREAD

**Issue by gbif-portal** [gbif/portal-feedback#2118]
**&quot;License: Unspecified&quot; due to unindexable file?**

The data file of this archive (`https://www.gbif.org/dataset/1acafd29-11ba-47c3-a84a-8ad7cb6e42b9:` `https://ipt.biodiversidad.co/sibm/archive.do?r=fito_serranilla)` cannot be indexed due to a violation of referential integrity in the data
 (`https://www.gbif.org/tools/data-validator/1562926582615).` This also appears to cause the license information that is provided in the EML file (CC-BY-NC) to be lost to the indexing process, so that the metadata stump gives the license as unspecified.

Is it correct that data errors in the data file also cause the EML metadata to not be reinterpreted at indexing time?

-----

User provided contact info: @ahahn-gbif
System: Firefox 68.0.0 / Windows 10.0.0
User: [See in registry](https://www.gbif.org/api/feedback/user/1ffee8ab794bc23e40b2308aac51d91d:3bc185cff7e5f703121b855fa8c6888cd61498f8aabc531f741ec925c5beacbf5ea620eb92f2abeaa5e87a38d7a2a67747f29f2b33454ac8749cc6d6d2407834)
Referer: https://www.gbif.org/dataset/1acafd29-11ba-47c3-a84a-8ad7cb6e42b9
Window size: width 1920 - height 1036
[API log](http://elk.gbif.org:5601/app/kibana?#/discover?_g=(refreshInterval:(display:Off,pause:!f,value:0),time:(from:'2019-07-24T07:50:14.794Z',mode:absolute,to:'2019-07-24T07:56:14.794Z'))&_a=(columns:!(_source),index:'prod-varnish-*',interval:auto,query:(query_string:(analyze_wildcard:!t,query:'response:%3E499')),sort:!('@timestamp',desc)))
[Site log](http://elk.gbif.org:5601/app/kibana?#/discover?_g=(refreshInterval:(display:Off,pause:!f,value:0),time:(from:'2019-07-24T07:50:14.794Z',mode:absolute,to:'2019-07-24T07:56:14.794Z'))&_a=(columns:!(_source),index:'prod-portal-*',interval:auto,query:(query_string:(analyze_wildcard:!t,query:'response:%3E499')),sort:!('@timestamp',desc)))
System health at time of feedback: OPERATIONAL

**Comment by MattBlissett** [comment]
Datasets registered via the "legacy" (IPT) API initially have an "unspecified" license, which is usually changed a minute later on the first successful ingestion of the published DWCA's EML.

If the DWCA is invalid, neither the metadata nor the data is ingested.

Described in https://github.com/gbif/registry/issues/71, this makes errors like failure to use a machine readable license immediately obvious, and we can catch them around the time of publication.

I'm not sure what's "correct" when this is a failure case anyway.

**Comment by MattBlissett** [comment]
There are two further cases where an invalid DWCA leaves the minimal dataset metadata without a license: https://www.gbif.org/dataset/search?type=OCCURRENCE&license=UNSPECIFIED

**Comment by MattBlissett** [comment]
And a fourth, which doesn't have a DWCA endpoint: https://www.gbif.org/dataset/c84b0aff-beb5-4dbb-a314-1eb32229ad68 (and is on a *very* old IPT).

**Comment by ahahn-gbif** [comment]
Agreed the data files need fixing. In the case starting this thread, though, I cannot quite spot the issue with the EML. The relevant section reads

`<intellectualRights>
    <para>This work is licensed under a 
        <ulink url="http://creativecommons.org/licenses/by-nc/4.0/legalcode">
            <citetitle>Creative Commons Attribution Non Commercial (CC-BY-NC) 4.0 License</citetitle>
        </ulink>.
    </para>
</intellectualRights>`

What am I missing?

**Comment by MattBlissett** [comment]
If the DWCA is invalid, processing stops before the EML is ingested. The rest of the metadata (taxonomic scope etc) is also in the EML, but not on the dataset page.

**Comment by ahahn-gbif** [comment]
I understand the fact that this is what the current workflow does, and will remain that way. Just raising that, from an outside perspective, it does not quite look plausible that there is a registered and functional EML endpoint at https://ipt.biodiversidad.co/sibm/eml.do?r=fito_serranilla, yet the index metadata do not represent the content. Agree that the data issue needs to be fixed anyway. Using the license content as an issue flag is a bit of a by-product.

**Comment by ahahn-gbif** [comment]
#2138
